text/0010-design-libraries.md

```diff
@@ -0,0 +1,47 @@
+# The Styleguide and Design Libraries
+- Start Date: 2019/06/08
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/10
+
+## Decision, including impact on distributions
+The [OpenMRS styleguide](/text/0008-styleguide.md) does not use a third party design library's code (such as Twitter Bootstrap,
+Material, Semantic UI). Instead, the styleguide is built from scratch in collaboration with UX designers in the OpenMRS community.
+UX principles and design elements are liberally borrowed from other design systems and libraries, but the code for the
+OpenMRS styleguide is written by the OpenMRS community.
+
+Distributions should use the OpenMRS styleguide as the primary source of making a consistent user experience, look and feel,
+layout, and base-level components. In addition to the OpenMRS styleguide, distributions may use design libraries if they prefer to.
+However, it is not necessary (nor encouraged) to do so.
+
+## Definition
+A design library is a third-party javascript library that provides pre-written css and javascript components. Design libraries
+are installed into code and are often used by the frontend community for base-level components (such as buttons, form elements,
+and tooltips).
+
+At the time of this writing, [Material UI](https://material-ui.com/) (from Google), [Twitter Bootstrap](https://getbootstrap.com/),
+and [Semantic UI](https://semantic-ui.com/) are a few of the popular design libraries.
+
+A custom build of a design library is a customized version of such a library. Customizations often include things like custom
+css class naming, additional css class definitions, and custom branding variables. Design libraries often provide a
+well-supported way to create a custom build of their library. Organizations can use custom builds instead of the generic version
+of the library to make the design library their own.
+
+## Reason for decision
+- Upgrading design libraries is often very difficult because breaking changes span across an organization's entire
+  frontend codebase. Many organizations get stuck on a particular version of a design library and end up having to
+  rewrite much of their code to be able to upgrade or switch to a different design library.
+- Because of trying to be useful to all organizations in the world, design libraries are often big and bloated. They
```

Comment:
If the OpenMRS styleguide has a dedicated and judicious owner, perhaps perhaps perhaps we will be able to avoid having something that's big and bloated in addition to being less than state-of-the-art (OpenMRS having somewhat fewer professional engineering and design resources than Google and Twitter).

Reply:
Agreed.

The point I make in the bullet point is referring to if we adopted an extant third party design library, such as Bootstrap, that we would be inheriting their bloat.

Reply:
I can understand the reluctance given very few UX experts for home-grown. But I agree with Joel on dependency mess and imposition/tight-binding due to 3rd parties! If we keep the surface lean, css classes for essential/base elements, and if that can be overridden by css vars, then I think in the long run it will benefit all. However, I hope this does not create bottlenecks on experts. 

Also even with OpenMRS custom style lib, it does not stop others from using their own or mix with other libraries. If we document "how to override" the documented base styles of OpenMRS css libs (hopefully a small footprint), then this would be easier to maintain/override as well, in case of breaking changes of OpenMRS css lib or specific-upgrades of 3rd party libs.